tests/micro/arduino/conftest.py

```diff
@@ -48,13 +35,23 @@
 ).resolve()
 
 
+def arduino_micro_devices() -> dict:
+    sys.path.insert(0, str(TEMPLATE_PROJECT_DIR))
+    try:
+        import microtvm_api_server
+    finally:
+        sys.path.pop(0)
+
+    return microtvm_api_server.MICRO_DEVICES
```

Comment:
I'm ok with importing the list of devices directly here for now. But I consider that a kind of shortcut to the API, however currently there is no way to get it via the API info query. That's another change I'm planning to submit soon because it's necessary for the TVMC + micro target work. Hence, only as a reference, that's what I'm planning

https://github.com/gromero/tvm/commit/7ff1f17f31d30d806c1fd4a6cbf5e609766901e0

That change would allow something like:

```
import tvm.micro.project
```
and then query the boards via the API:

```
template = project.TemplateProject.from_directory(ZEPHYR_TEMPLATE_DIR)
boards = template.info()["zephyr_board"]
```

Anyways, I'm ok with importing directly for now.

Reply:
i don't favor doing this. can we use the ProjectOption/choices infrastructure if we want to forward this data? this preserves the API boundary.

Reply:
I'm with Andrew - is there any way this can be done by reading the choices from `ProjectOption`?

Reply:
Changed conftest to use choices from project options.